GDJS/GDJS/Extensions/Builtin/CommonInstructionsExtension.cpp

```diff
@@ -101,6 +103,76 @@ CommonInstructionsExtension::CommonInstructionsExtension() {
         return outputCode;
       });
 
+  GetAllEvents()["BuiltinCommonInstructions::Async"].SetCodeGenerator(
+      [](gd::BaseEvent& event_,
+         gd::EventsCodeGenerator& codeGenerator,
+         gd::EventsCodeGenerationContext& parentContext) {
+        gd::AsyncEvent& event = dynamic_cast<gd::AsyncEvent&>(event_);
+
+        const gd::String callbackFunctionName =
+            "asyncCallback" +
+            gd::String::From(codeGenerator.GenerateSingleUsageUniqueIdFor(
+                &event.GetInstruction()));
+
+        // Generate callback code
+        gd::EventsCodeGenerationContext callbackContext;
+        callbackContext.InheritsAsAsyncCallbackFrom(parentContext);
+
+        // Generate actions
+        gd::String actionsCode = codeGenerator.GenerateActionsListCode(
+            event.GetActions(), callbackContext);
+
+        // Generate subevents
+        if (event.HasSubEvents())  // Sub events
+        {
+          actionsCode += "\n{ //Subevents\n";
+          actionsCode += codeGenerator.GenerateEventsListCode(
+              event.GetSubEvents(), callbackContext);
+          actionsCode += "} //End of subevents\n";
+        }
+
+        // Compose the callback function and add outside main
+        const gd::String actionsDeclarationsCode =
+            codeGenerator.GenerateObjectsDeclarationCode(callbackContext);
+        const gd::String callbackCode = "function " + callbackFunctionName +
+                                        "(runtimeScene, asyncObjectsList) {\n" +
```

Comment:
Note for me (or you): changing this to use `GenerateEventsParameters` fix the tests (by passing the proper arguments).